Code/Tools/FBuild/FBuildTest/Tests/TestIf.cpp

```diff
@@ -115,6 +122,39 @@ void TestIf::IfNotSetFunctionFalse() const
 //------------------------------------------------------------------------------
 void TestIf::IfFunctionBool() const
 {
+    // Literals
+    TEST_EXP_TRUE( "", "true" );
+    TEST_EXP_FALSE( "", "false" );
+
+    // Unary
+    TEST_EXP_TRUE( "", "!false" );
+    TEST_EXP_FALSE( "", "!true" );
+    
+    // Binary
+    TEST_EXP_TRUE( "", "false != true" );
+
+    // Unary and Binary
+    TEST_EXP_TRUE( "", "!false == true" );
+    TEST_EXP_TRUE( "", "!true == false" );
+
+    // Vars
+    TEST_EXP_TRUE( ".Bool = true", ".Bool" );
+    TEST_EXP_TRUE( ".Bool = true", ".Bool != false" );
+    TEST_EXP_TRUE( ".Bool = true", "false != .Bool" );
+    TEST_EXP_FALSE( ".Bool = true", ".Bool != .Bool" );
+
+    // Compound Exps
+    TEST_EXP_TRUE(R"(
```

Comment:
Several versions of the MSVC compiler have bugs that result in bad debug info when multi-line string literals are used.
For this reason, FASTBuild explicitly avoids their use (you'll note other tests excplicitly specifying line breaks).

It would be best to change these to avoid these bugs.

Reply:
Fixed in 98ef95275848f20cea086e8c02cc3e13c8657003.